formatter/formatter.go

```diff
@@ -15,14 +15,30 @@ type Formatter interface {
 	FormatQueryDocument(doc *ast.QueryDocument)
 }
 
-func NewFormatter(w io.Writer) Formatter {
-	return &formatter{writer: w}
+type FormatterOption func(*formatter)
+
+func WithSpaceIndent(tabSize int) FormatterOption {
```

Comment:
Since indentation can be either space or tab (as it’s today), I’ll suggest changing method signature to:
```suggestion
func WithIndent(indent string) FormatterOption {
```

Like json.MarshalIndent. What do you think?